src/index.js

```diff
@@ -1,13 +1,29 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import React from "react";
+import ReactDOM from "react-dom/client";
+import "./index.css";
+import App from "./App";
+import reportWebVitals from "./reportWebVitals";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import ErrorPage from "./error-page";
+import Invoice from "./pages/invoice.js";
 
-const root = ReactDOM.createRoot(document.getElementById('root'));
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <App />,
+    errorElement: <ErrorPage />,
+  },
+  {
+    path: "invoices/add",
+    element: <Invoice />,
+    errorElement: <ErrorPage />,
+  },
+]);
+
+const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
   <React.StrictMode>
-    <App />
+    <RouterProvider router={router} />
```

Comment:
ołkej, można i tak, ALE, przenieś to do app.js i umieść nad RouterProviderem całą resztę interfejsu która jest wspolna- np. navbar, albo jakiś footer jeśli będzie, tak żeby tylko "środek" strony się zmieniał

Reply:
Aczkolwiek, typowo robi się to inaczej- co nie znaczy że to jest źle. Mianowicie, zamias tego createBrowerRouter, mozesz uzyc takiego zwyklego inline jsx'a, typu <BrowerRouter><Route /> i tam sobie dać wszystkie routy jakie chcesz, a na koniec dać taki co zwróci 404 jak zaden inny nie będzie pasował

Reply:
```
<BrowserRouter>
   <Routes>
       <Route path="/" element={<HomePage />} />
       <Route path="/contact" element={<ContactPage />} />

       {/* 404 rounte */}
       <Route path="*" element={<NotFoundPage />} />
   </Routes>
</BrowserRouter>
```

Reply:
Zastosowałam `BrowserRouter`